test/testdrive/aws-connection.td

```diff
@@ -0,0 +1,77 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Tests for AWS connections
+
+$ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+ALTER SYSTEM SET enable_aws_connection = true;
+ALTER SYSTEM SET enable_connection_validation_syntax = true;
+
+# Testing with assume role
+> CREATE CONNECTION aws_assume_role
+  TO AWS ( ASSUME ROLE ARN 'assume-role', ASSUME ROLE SESSION NAME 'session-name' );
+
+$ set-from-sql var=conn_id
+SELECT id FROM mz_connections WHERE name = 'aws_assume_role';
+
+> SELECT * FROM mz_aws_connections WHERE id = '${conn_id}';
+"${conn_id}" "<null>" "<null>" "<null>" "<null>" "assume-role" "session-name" "arn:aws:iam::123456789000:role/MaterializeConnection" "mz_eb5cb59b-e2fe-41f3-87ca-d2176a495345_${conn_id}" "{\"Statement\":[{\"Action\":\"sts:AssumeRole\",\"Condition\":{\"StringEquals\":{\"sts:ExternalId\":\"mz_eb5cb59b-e2fe-41f3-87ca-d2176a495345_${conn_id}\"}},\"Effect\":\"Allow\",\"Principal\":{\"AWS\":\"arn:aws:iam::123456789000:role/MaterializeConnection\"}}],\"Version\":\"2012-10-17\"}"
```

Comment:
This would be a good moment to use the syntax that lets you include/assert column headers in the result: https://github.com/MaterializeInc/materialize/blob/7bdcf673c18bfa14ea14120dd76a01e712c74691/test/testdrive/array.td#L14-L15

That should make it easier to follow which columns are which! Also you don't need to wrap each column in quotes. Just the first one (because it starts with $).